common-test/src/main/java/com/duckduckgo/app/global/api/InMemorySharedPreferences.kt

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.global.api
+
+import android.content.SharedPreferences
+
+class InMemorySharedPreferences : SharedPreferences, SharedPreferences.Editor {
```

Comment:
Added this class to easily fake shared preferences. The usage of this is limited right now. We can create a subsequent task to increase test coverage when possible